### PR TITLE
Fix support for x-errors in OAS

### PR DESCRIPTION
**Pull request by kong-apiops**
@mheap made additional changes to the x-error rendering in this PR. Please review

### CODE REVIEW

File: app/_plugins/generators/oas_definition/product.rb [hunk 1]
```diff
@@ -4,28 +4,54 @@ module OasDefinition
...
 
     private
 
-    def generate_product_pages!(product, data)
+    def oas_x_errors(frontmatter)
+      spec_path = api_spec_path(frontmatter)
+      return nil unless spec_path
+
+      oas = load_api_file(spec_path)
+      raise "Could not load #{spec_path}" unless oas
```
Comment (fabianrbz) [inline app/_plugins/generators/oas_definition/product.rb:34]: @mheap do we need all these implementation changes?
I think we could achieve the same thing by adding this check before [this](https://github.com/Kong/docs.konghq.com/pull/7494/files#diff-86ae4e7231f5c58a4b243dc4ff98788251b4259f8a064acee8752bdb08554049L80) line.

Comment (mheap) [reply app/_plugins/generators/oas_definition/product.rb:34]: At that point the `site.pages << generate_error_page` has run. Is there a way to abort page generation from within `ErrorPage`?

Comment (fabianrbz) [reply app/_plugins/generators/oas_definition/product.rb:34]: got it, we no longer support `generateErrorsPage` then?

**Review by fabianrbz — APPROVED** [review]

**Comment by fabianrbz** [comment]
@mheap looks like the branch was force-pushed by the bot :(

File: app/_plugins/generators/oas_definition/product.rb (hunk 1 above)
Comment (mheap) [reply app/_plugins/generators/oas_definition/product.rb]: Nope. I realised that it was manual metadata in a file that's autogenerated from a tool

**Comment by mheap** [comment]
@fabianrbz It was just for testing purposes in this repo. I need to fix the upstream build to push it down

**Pull request merged**